### PR TITLE
Redo "Revert "Deprecate String>>asDate"

### PR DESCRIPTION
**Pull request by guillep**
Redo https://github.com/pharo-project/pharo/pull/15152, originally reverted because there are still broken tests.

Date>>asDate is ambiguous regarding how to interpret the day, month and year fields. Sometimes it interprets them in some order and sometimes in other.

```smalltalk
'1-13-04' asDate => 13 january 2004
```

```smalltalk
'13-1-04' asDate => 13 january 2004
```

Recently an exception was added in one of the cases above, but the ambiguity remains.
We propose with @guillep to deprecate it in favor of `#readFrom:pattern:` with an explicit pattern.

We deprecate methods: 
- String>>asDate,
- Date>>fromString, and 
- Date>>readFrom.
Remove related tests on the deprecated methods.
Use Date>>#readFrom:pattern: specifying a concrete pattern instead + all other.

We also added an option into the pattern parser to continue parsing using the continueAfterParsing method.

### CODE REVIEW

File: src/Calypso-SystemQueries/Project.extension.st [hunk 1]
```diff
@@ -1,7 +1,7 @@
 Extension { #name : 'Project' }
 
 { #category : '*Calypso-SystemQueries' }
-Project class >> convertToCalypsoBrowserItem: aProject [
```
Comment (guillep) [inline src/Calypso-SystemQueries/Project.extension.st:4]: I got this change because of a merge. @jecisc do you know why?

Comment (jecisc) [reply src/Calypso-SystemQueries/Project.extension.st:4]: I have no idea where those "classSide" are coming from.. I've see it in the past in some projects but I never discovered why. 

I never do my merges with Pharo, maybe it's the Pharo merger that uses another printer?

**Comment by guillep** [comment]
Strange, I have lots of tests failing that look unrelated to my changes, and look more related to the handling of protocols and extension methods, any idea?

imagen

**Comment by guillep** [comment]
They work in my local image based on build 1020.

Pharo 12.0.0
Build information: Pharo-12.0.0+build.1020.sha.373a7ce1fe48d19844129ab9086329fcf758ce50 (64 Bit)

**Comment by jecisc** [comment]
This is a known failure I'm working on that appears really recently: https://github.com/pharo-project/pharo/issues/15282

The error is really strange because it happens when updating the users of a trait, but the trait is generated in the tests and has no user (: 

I'll try to fix it this week

**Comment by guillep** [comment]
Oh! Not my fault! Then I can remove the WIP from this PR :)

**Comment by guillep** [comment]
Why closing it instead of merging it?

**Pull request merged**